lib/Data/Validate/Sanctions/Fetcher.pm

```diff
@@ -425,6 +429,91 @@ sub _eu_xml {
     };
 }
 
+sub _unsc_xml {
+    my ($xml_content) = @_;
+
+    # Preprocess the XML content to escape unescaped ampersands
+    $xml_content =~ s/&(?!(?:amp|lt|gt|quot|apos);)/&amp;/g;
+    my $data = xml2hash($xml_content, array => ['INDIVIDUAL', 'INDIVIDUAL_ALIAS', 'INDIVIDUAL_ADDRESS', 'INDIVIDUAL_DATE_OF_BIRTH', 'INDIVIDUAL_PLACE_OF_BIRTH', 'INDIVIDUAL_DOCUMENT'])->{CONSOLIDATED_LIST};
+
+    # Extract the dateGenerated attribute from the first line of the XML content
+    my ($date_generated) = $data->{'-dateGenerated'};
+    die "Corrupt data. Release date is missing\n" unless $date_generated;
+
+    # Convert the dateGenerated to epoch milliseconds
+    my $publish_epoch = _date_to_epoch($date_generated // '');
+
+    my $dataset = [];
+
+    for my $individual (@{$data->{'INDIVIDUALS'}->{'INDIVIDUAL'}}) {
+        my %entry;
+
+        $entry{first_name}           = $individual->{'FIRST_NAME'};
+        $entry{second_name}          = $individual->{'SECOND_NAME'};
+        $entry{third_name}           = $individual->{'THIRD_NAME'}           // '';
+        $entry{fourth_name}          = $individual->{'FOURTH_NAME'}          // '';
+        $entry{name_original_script} = $individual->{'NAME_ORIGINAL_SCRIPT'} // '';
+
+        my @names = (
+            $entry{first_name}           // '',
+            $entry{second_name}          // '',
+            $entry{third_name}           // '',
+            $entry{fourth_name}          // '',
```

Comment:
I've noticed that in a couple of other places the first_name, second_name and etc, are combined into a single string.
_ofac_xml
_hmt_csv

Reply:
We can do that but then some individuals have two names present in the first name section e.g. MOHAMMAD NAIM. If we search with MOHAMMAD NAIM we find it in this way, if we join then the array is going to have MOHAMMAD NAIM BARICH instead. Do you think we should have it like that? or Is this way fine?